### PR TITLE
Some minor updates

### PR DESCRIPTION
**Pull request by rudro**
- Carry file path around to a few new places
- Prep for new RegEx class in Swift 5.7
- Deal with internal properties
- Track leaf components

### CODE REVIEW

File: Generator/Sources/NeedleFramework/Models/Property.swift [hunk 1]
```diff
@@ -23,6 +23,18 @@ struct Property: Hashable {
     let name: String
     /// The property type `String`.
     let type: String
+    /// If this properlty is internal
```
Comment (wbond) [inline Generator/Sources/NeedleFramework/Models/Property.swift:26]:
```suggestion
    /// If this property is internal
```

File: Generator/Sources/NeedleFramework/Parsing/Pluginized/Processors/NonCoreComponentLinker.swift [hunk 2]
```diff
@@ -39,13 +39,19 @@ class NonCoreComponentLinker: Processor {
         var nonCoreMap = [String: ASTComponent]()
         for nonCoreComponent in nonCoreComponents {
             nonCoreMap[nonCoreComponent.name] = nonCoreComponent
+            if nonCoreComponent.isLeaf {
+                nonCoreComponent.isLeaf = false
+            }
```
Comment (wbond) [inline Generator/Sources/NeedleFramework/Parsing/Pluginized/Processors/NonCoreComponentLinker.swift:44]: It wasn't clear to me what this was accomplishing. How would `isLeaf` be set incorrectly here? Perhaps a comment explaining the situation would be helpful.

File: Generator/Sources/NeedleFramework/Parsing/Pluginized/Processors/NonCoreComponentLinker.swift [hunk 3]
```diff
@@ -39,13 +39,19 @@ class NonCoreComponentLinker: Processor {
...
         }
 
         for pluginizedComponent in pluginizedComponents {
             guard let nonCoreComponent = nonCoreMap[pluginizedComponent.nonCoreComponentType] else {
                 throw GenericError.withMessage("Cannot find \(pluginizedComponent.data.name)'s non-core component with type name \(pluginizedComponent.nonCoreComponentType)")
             }
 
+            if pluginizedComponent.data.isLeaf {
+                pluginizedComponent.data.isLeaf = false
+            }
```
Comment (wbond) [inline Generator/Sources/NeedleFramework/Parsing/Pluginized/Processors/NonCoreComponentLinker.swift:54]: Similar to above, I didn't understand this change.

**Review by wbond — CHANGES_REQUESTED** [review]
Overall the changes seemed to make sense, just a typo and a question about two small bits.

File: Generator/Sources/NeedleFramework/Parsing/Pluginized/Processors/NonCoreComponentLinker.swift (hunk 3 above)
Comment (rudro) [reply Generator/Sources/NeedleFramework/Parsing/Pluginized/Processors/NonCoreComponentLinker.swift]: Added a comment

File: Generator/Sources/NeedleFramework/Parsing/Pluginized/Processors/NonCoreComponentLinker.swift (hunk 2 above)
Comment (rudro) [reply Generator/Sources/NeedleFramework/Parsing/Pluginized/Processors/NonCoreComponentLinker.swift]: Added a comment

**Review by wbond — APPROVED** [review]
LGTM

**Comment by rudro** [comment]
Merged without the license check passing as it was taking too long to run.

**Pull request merged**